packages/story-editor/src/components/library/panes/media/common/insertionMenu.js

```diff
@@ -0,0 +1,200 @@
+import useStory from '../../../../../app/story/useStory';
+import useNestedRovingTabIndex from '../../shared/hooks/useNestedRovingTabIndex';
+
+const AddButton = styled(Button).attrs({ variant: BUTTON_VARIANTS.ICON })`
+  display: flex;
+  align-items: center;
+  position: absolute;
+  top: calc(50% - 16px);
+  right: calc(50% - 16px);
+  color: ${({ theme }) => theme.colors.fg.primary};
+  border-radius: 100%;
+  width: 28px;
+  height: 28px;
+  opacity: ${({ display }) => (display ? '1' : '0')};
+`;
+
+const IconContainer = styled.div`
+  height: 32px;
+  width: 32px;
+`;
+
+const DropDownContainer = styled.div`
+  margin-top: 10px;
+  min-width: 160px;
+`;
+
+const MenuContainer = styled.div`
+  z-index: 1;
+`;
+
+const menuStylesOverride = css`
+  min-width: 100px;
+  margin-top: 0;
+  li {
+    display: block;
+  }
+`;
+
+/**
+ * Get a More icon that displays a dropdown menu on click.
+ *
+ * @param {Object} props Component props.
+ * @param {Object} props.resource Selected media element's resource object.
+ * @param {boolean} props.display Whether the more icon should be displayed.
+ * @param {Function} props.onInsert Callback for inserting media.
+ * @param {number} props.width Media width.
+ * @param {number} props.index Element index in the gallery.
+ * @return {null|*} Element or null if should not display the More icon.
+ */
+function InsertionMenu({ resource, display, onInsert, width, index }) {
+  const insertButtonRef = useRef();
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const onMenuOpen = useCallback((e) => {
+    e.stopPropagation();
+    setIsMenuOpen(true);
+  }, []);
+  const onMenuCancelled = useCallback(() => setIsMenuOpen(false), []);
+
+  const { currentBackgroundId, combineElements } = useStory((state) => ({
+    currentBackgroundId: state.state.currentPage?.elements?.[0]?.id,
+    combineElements: state.actions.combineElements,
+  }));
+
+  const { type, poster } = resource;
+  const insertLabel = ['image', 'gif'].includes(type)
+    ? __('Insert image', 'web-stories')
+    : __('Insert video', 'web-stories');
+  const options = [
+    {
+      group: [
+        { label: insertLabel, value: 'insert' },
+        {
+          label: __('Add as background', 'web-stories'),
+          value: 'addBackground',
+        },
+      ],
+    },
+  ];
+
+  const handleCurrentValue = (evt, value) => {
+    const thumbnailUrl = poster
+      ? poster
+      : getSmallestUrlForWidth(width, resource);
+    const newElement = getElementProperties(resource.type, {
+      resource,
+    });
+    switch (value) {
+      case 'insert':
+        onInsert(resource, thumbnailUrl);
+        break;
+      case 'addBackground':
+        combineElements({
+          firstElement: newElement,
+          secondId: currentBackgroundId,
+        });
+        break;
+      default:
+        break;
+    }
+  };
+
+  useNestedRovingTabIndex({ ref: insertButtonRef }, [], 3);
```

Comment:
What is `3` here?  I'm a broken record here, but giving this a name would help when reading through the code to know what it does without having to trace it back to the hook.